certifier_service/certlib/certlib_support.go

```diff
@@ -1781,8 +1793,41 @@ func GetSubjectKey(cert *x509.Certificate) *certprotos.KeyMessage {
 			fmt.Printf("GetSubjectKey: Can't internal ecc public key\n")
 			return nil
 		}
+
+		// Look for AMD VCEK cert extensions if they exist
+		oidBlSPL := "1.3.6.1.4.1.3704.1.3.1"
+		oidTeeSPL := "1.3.6.1.4.1.3704.1.3.2"
+		oidSnpSPL := "1.3.6.1.4.1.3704.1.3.3"
+		oidUcodeSPL := "1.3.6.1.4.1.3704.1.3.8"
```

Comment:
```suggestion
		oidBlSPL := "1.3.6.1.4.1.3704.1.3.1"
		oidTeeSPL := "1.3.6.1.4.1.3704.1.3.2"
		oidSnpSPL := "1.3.6.1.4.1.3704.1.3.3"
		oidUcodeSPL := "1.3.6.1.4.1.3704.1.3.8"
		oidHwId := "1.3.6.1.4.1.3704.1.4"
```

Looks like this is missing the Hardware ID, which is a critical part of the extension, as it matches the physical chip id of the secure processor belonging to a given node. You will likely want to add a slice comparison function/method to perform this comparison, as it is an array of bytes.

Reply:
Thanks for the info! I didn't realize it's the chip id in the report. Will add that.